Update users subscriptions

balajisubr opened the pull request:
https://github.com/electology/approval_frame/issues/124
Dependencies: https://github.com/electology/approval_frame/pull/122 to be merged

balajisubr commented:
@rupalkhilari please take a look at this for updating Subscription model with mailchimp data. The cron entry is using python to run the file.  I can look into separating this method out into a separate file. Let me know if you have any questions

rupalkhilari commented:
Thanks @balajisubr, I will take a look at this today and get back to you.

rupalkhilari commented on approval_frame/approval_frame_cron:
```diff
@@ -0,0 +1 @@
+0 0 * * * python mailchimp_api.py
```
Is this working for you ? We need to run this script in the django context. I think following django-extensions runscript for this might be good. http://django-extensions.readthedocs.io/en/latest/runscript.html . The structure of having a scripts directory in the top level would also be good instead of using a mail method in the api module. Let me know what you  think.

rupalkhilari commented on approval_frame/mailchimp_api.py:
```diff
@@ -8,3 +11,49 @@ def get_mailchimp_api():
+        if user_present == False:
+            user.subscription_set.first().delete()
+
+if __name__ == "__main__":
```
It would be good if you could move this to a scripts/ directory and attempt to call it using runscript from django-extensions. http://django-extensions.readthedocs.io/en/latest/runscript.html

balajisubr commented:
@rupalkhilari I have made the script to run as part of Django Extensions runscript. I have moved it to a separate file and also updated the cron. Are you planning to create a symlink to this cron from the systems' cron we deploy ? Let me know if you have any other questions.

rupalkhilari commented on approval_frame/approval_frame_cron:
```diff
@@ -1 +1 @@
-0 0 * * * python mailchimp_api.py
+0 0 * * * python manage.py runscript approval_frame.scripts.update_user_subscriptions
```
I think you should remove this cron file. As we would need to source the virtual env on the server before running the script, I think we could directly add it in the system/user crontab on the server.

balajisubr commented:
@rupalkhilari I have made the requested changes. Is there a staging environment where we can test this? Also I will look into optimizing the script along with changes for https://github.com/electology/approval_frame/issues/108

rupalkhilari commented:
There is no staging environment yet. You should be able to test it out on your linux box. I shall be deploying the current updates in production in a day or two.
